go-controller/pkg/ovn/controller/services/lb_config.go

```diff
@@ -41,45 +41,42 @@ type lbConfig struct {
 	hasNodePort bool
 }
 
-func (c *lbConfig) makeNodeSwitchTargetIPs(node *nodeInfo, epIPs []string) (targetIPs []string, changed bool) {
-	targetIPs = epIPs
+func (c *lbConfig) makeNodeSwitchTargetIPs(service *v1.Service, node *nodeInfo, endpoints []discovery.Endpoint) (targetIPs []string, changed bool) {
+	allTargetIPs := util.GetEligibleEndpointAddresses(endpoints, service)
+	targetIPs = allTargetIPs
 	changed = false
-
-	if c.externalTrafficLocal {
+	if c.externalTrafficLocal || c.internalTrafficLocal {
 		// for ExternalTrafficPolicy=Local, remove non-local endpoints from the router/switch targets
 		// NOTE: on the switches, filtered eps are used only by masqueradeVIP
-		targetIPs = util.FilterIPsSlice(targetIPs, node.nodeSubnets(), true)
-	}
-
-	if c.internalTrafficLocal {
 		// for InternalTrafficPolicy=Local, remove non-local endpoints from the switch targets only
-		targetIPs = util.FilterIPsSlice(targetIPs, node.nodeSubnets(), true)
+		targetIPs = util.GetLocalEligibleEndpointAddresses(endpoints, service, node.name)
 	}
 
 	// We potentially only removed stuff from the original slice, so just
-	// comparing lenghts is enough.
-	if len(targetIPs) != len(epIPs) {
+	// comparing lengths is enough.
+	if len(targetIPs) != len(allTargetIPs) {
 		changed = true
 	}
 	return
 }
 
-func (c *lbConfig) makeNodeRouterTargetIPs(node *nodeInfo, epIPs []string, hostMasqueradeIP string) (targetIPs []string, changed bool) {
-	targetIPs = epIPs
+func (c *lbConfig) makeNodeRouterTargetIPs(service *v1.Service, node *nodeInfo, endpoints []discovery.Endpoint, hostMasqueradeIP string) (targetIPs []string, changed, zeroRouterLocalEndpoints bool) {
```

Comment:
`zeroRouterLocalEndpoints` not used?

Reply:
ah ops, it's used in a downstream-only commit: https://github.com/openshift/ovn-kubernetes/blob/master/go-controller/pkg/ovn/controller/services/lb_config.go#L78
I'll remove it here.